### PR TITLE
Docs/orc write dyozie

### PR DESCRIPTION
**Pull request by dyozie**

### CODE REVIEW

File: docs/content/hdfs_orc.html.md.erb [hunk 1]
```diff
@@ -207,3 +263,26 @@ Procedure:
...
     (1 row)
     ```
+
+## <a id="write_example"></a>Example: Writing to an ORC File on HDFS
+
+1. Use the `hdfs:orc` profile to create a writable external table that references `sample_orc`:
+
+    ``` sql
+    postgres=# CREATE WRITABLE EXTERNAL TABLE write_to_sample_orc (location text, month text, number_of_orders int, total_sales numeric(10,2), items_sold text[] )
+        LOCATION ('pxf://data/pxf_examples/sampledata.orc?PROFILE=hdfs:orc')
```
Comment (dyozie) [inline docs/content/hdfs_orc.html.md.erb:273]: @bradfordb-vmware this command intends to create a writable external table that references and writes to the `sample_orc` table created in the previous read example. I don't see how there's any connection to that table if we change the path from `sampledata.orc`.  I've added some clarifying text following that this references a directory containing segment files, but please double-check this procedure for technical accuracy.  Maybe what's intended here isn't possible with the current syntax?

File: docs/content/hdfs_orc.html.md.erb [hunk 2]
```diff
@@ -68,16 +80,59 @@ PXF supports only the list ORC compound type, and only for a subset of the ORC s
 | array\<int> | Integer[] |
 | array\<bigint> | Bigint[] |
 
+### <a id="write_map"></a>Write Mapping
+
+PXF uses the following data type mapping when writing ORC data:
```
Comment (dyozie) [inline docs/content/hdfs_orc.html.md.erb:85]: @ashuka24 please double-check accuracy of these new tables for write support.  Thanks!

File: docs/content/hdfs_orc.html.md.erb (hunk 1 above)
Comment (bradfordb-vmware) [reply docs/content/hdfs_orc.html.md.erb:273]: In the read example, a single file called `sampledata.orc` is created and then placed in HDFS. PXF can read a single file from HDFS. However, when PXF writes data to HDFS it writes one file per segment. So for writable tables, the path given in the location URL cannot be a file -- PXF will error out:

```sql
INSERT INTO write_to_sample_orc VALUES ( 'Frankfurt', 'Mar', 777, 3956.98, '{"winter socks",pants,boots}' );
ERROR:  PXF server error : Destination exists and is not a directory: /pxf-data/pxf_examples/sampledata.orc  (seg0 127.0.1.1:6000 pid=498533)
HINT:  Check the PXF logs located in the '/home/bboyle/pxf-base/logs' directory on host 'alpine' or 'set client_min_messages=LOG' for additional details.
```

I think we'd want to use something like `/pxf-data/pxf_examples/orc_example` for the path in the location URL of the readable and writable table. The file `sampledata.orc` can then be copied into that location.

```sql
CREATE EXTERNAL TABLE sample_orc(location text, month text, num_orders int, total_sales numeric(10,2), items_sold text[]) LOCATION ('pxf://data/pxf_examples/orc_example?PROFILE=hdfs:orc') FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');

CREATE WRITABLE EXTERNAL TABLE write_to_sample_orc (location text, month text, number_of_orders int, total_sales numeric(10,2), items_sold text[] ) LOCATION ('pxf://data/pxf_examples/orc_example?PROFILE=hdfs:orc') FORMAT 'CUSTOM' (FORMATTER='pxfwritable_export');
```

And update the HDFS command:

```bash
hdfs dfs -put /tmp/sampledata.orc /data/pxf_examples/orc_example
```

File: docs/content/hdfs_orc.html.md.erb [hunk 3]
```diff
@@ -207,3 +263,26 @@ Procedure:
...
+    postgres=# CREATE WRITABLE EXTERNAL TABLE write_to_sample_orc (location text, month text, number_of_orders int, total_sales numeric(10,2), items_sold text[] )
+        LOCATION ('pxf://data/pxf_examples/sampledata.orc?PROFILE=hdfs:orc')
+      FORMAT 'CUSTOM' (FORMATTER='pxfwritable_export');
+    ```
+
+1. Write a few records to segment files in the `sampledata.orc` directory by inserting to the `write_to_sample_orc` table:
+   
+    ``` sql
+postgres=# INSERT INTO write_to_sample_orc VALUES ( 'Frankfurt', 'Mar', 777, 3956.98, {"winter socks","pants",boots} );
+postgres=# INSERT INTO write_to_sample_orc VALUES ( 'Cleveland', 'Oct', 3218, 96645.37, {"long-sleeved shirts",hats} );
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_orc.html.md.erb:281]:
```suggestion
postgres=# INSERT INTO write_to_sample_orc VALUES ( 'Frankfurt', 'Mar', 777, 3956.98, '{"winter socks","pants",boots}' );
postgres=# INSERT INTO write_to_sample_orc VALUES ( 'Cleveland', 'Oct', 3218, 96645.37, '{"long-sleeved shirts",hats}' );
```

File: docs/content/hdfs_orc.html.md.erb [hunk 4]
```diff
@@ -207,3 +263,26 @@ Procedure:
...
+    ```
+
+1. Recall that Greenplum Database does not support directly querying a writable external table. Query the `sample_orc` table to read the data you added to files in the `sampledata.orc` directory:
+
+    ``` sql
+    postgres=# SELECT * FROM sample_orc ORDER BY number_of_orders;
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_orc.html.md.erb:287]:
```suggestion
    postgres=# SELECT * FROM sample_orc ORDER BY num_orders;
```

File: docs/content/hdfs_orc.html.md.erb [hunk 5]
```diff
@@ -207,3 +263,26 @@ Procedure:
      Prague    | Jan   |        101 |     4875.33 | {boots,hats}
     (1 row)
     ```
+
+## <a id="write_example"></a>Example: Writing to an ORC File on HDFS
+
+1. Use the `hdfs:orc` profile to create a writable external table that references `sample_orc`:
+
+    ``` sql
+    postgres=# CREATE WRITABLE EXTERNAL TABLE write_to_sample_orc (location text, month text, number_of_orders int, total_sales numeric(10,2), items_sold text[] )
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_orc.html.md.erb:272]:
```suggestion
    postgres=# CREATE WRITABLE EXTERNAL TABLE write_to_sample_orc (location text, month text, num_orders int, total_sales numeric(10,2), items_sold text[] )
```

File: docs/content/objstore_orc.html.md.erb [hunk 6]
```diff
@@ -46,25 +47,34 @@ The specific keywords and values used in the Greenplum Database [CREATE EXTERNAL
...
     CREATE EXTERNAL TABLE sample_orc( location TEXT, month TEXT, num_orders INTEGER, total_sales NUMERIC(10,2), items_sold TEXT[] )
-      LOCATION('pxf://BUCKET/pxf_examples/sampledata.orc?PROFILE=s3:orc&SERVER=s3srvcfg')
+      LOCATION('pxf://BUCKET/pxf_examples/sample-orc-data?PROFILE=s3:orc&SERVER=s3srvcfg')
     FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
     ```
 
+- Using the `CREATE WRITABLE EXTERNAL TABLE` syntax and `LOCATION` keywords and settings described above for the writable external table. For example, if your server name is `s3srvcfg`:
+
+    ``` sql
+    CREATE WRITABLE EXTERNAL TABLE write_to_sample_orc (location TEXT, month TEXT, number_of_orders INT, total_sales NUMERIC(10,2), items_sold TEXT[])
```
Comment (bradfordb-vmware) [inline docs/content/objstore_orc.html.md.erb:76]:
```suggestion
    CREATE WRITABLE EXTERNAL TABLE write_to_sample_orc (location TEXT, month TEXT, num_orders INT, total_sales NUMERIC(10,2), items_sold TEXT[])
```

**Review by bradfordb-vmware — APPROVED** [review]

File: docs/content/hdfs_orc.html.md.erb (hunk 1 above)
Comment (dyozie) [reply docs/content/hdfs_orc.html.md.erb:273]: Thanks for that clarification - if I understand correctly, then, the table created in **Example: Writing to an ORC File on HDFS** doesn't have any relationship to the table from the previous read example.  Is that correct?  If that's true then the final step of the write procedure should probably be removed or changed somehow. If GPDB doesn't support directly querying a writable external table I'm not sure how to show the results that were inserted in step 2.

File: docs/content/hdfs_orc.html.md.erb (hunk 2 above)
Comment (ashuka24) [reply docs/content/hdfs_orc.html.md.erb:85]: I apologize for taking a while to get back to you. Could we possibly reverse the table? 
i.e: 
```
| PXF/Greenplum Data Type | ORC Logical Type | ORC Physical Type |
```

also, we need to add UUID and timestamp with timezone to the write table mapping

the mapping for those is as follows: 
```
| PXF/Greenplum Data Type | ORC Logical Type | ORC Physical Type |
UUID | string | byte[] |
timestamp with timezone | timestamp with local time zone | timestamp |
```

Comment (dyozie) [reply docs/content/hdfs_orc.html.md.erb:85]: Thanks @ashuka24 - I merged this change in a separate PR, but the updated table is now at https://docs.vmware.com/en/VMware-Tanzu-Greenplum-Platform-Extension-Framework/6.4/tanzu-greenplum-platform-extension-framework-6-4/GUID-hdfs_orc.html?hWord=N4IghgNiBcIKpwJIBEQF8g#write-mapping-4

**Pull request merged**